### DISCUSSION TITLE
trivia brb screen

### DISCUSSION THREAD

**Discussion by inkydom** [TASagent/TASagentTwitchBotCore#3] [Ideas]
#### concept

I've been thinking about finding appropriately licensed js games which are turn based and incorporating them into a new browser source for start screen / brb engagement. I think the 'easiest' starting point would be trivia of which I have found a trivia question api: https://opentdb.com/api_config.php . I'm opening this discussion as an opportunity to bounce ideas or get feedback, no response required of course.

#### primary concern(s)

Mainly I need to do my own research to see how the license works with GPL3 / if it's compatible at all. 

#### Some of my thoughts:

1. As a new browser source, I'd probably use vuejs (but am open to react or another framework), mainly I like the structure and reactivity a bit better than jquery. Also as a new browser source it can be entirely isolated.
2. I probably want to store questions as they are retrieved, with a nullable timestamp in the db for 'asked_on', query for asked questions, and when we run out of unasked questions to ping the trivia database for more. This way we can also store analytics on correct vs incorrectly answered questions.
3. Eventually a points systems could be incorporated, but probably not with the initial launch
4. I'm thinking syntax for the commands:
!set trivia [start/stop]
!trivia summary (shows top 3 place winners-- maybe a later feature)
!trivia skip (skip a question mod/admin, if it's inappropriate)

**Comment by inkydom** [comment]
seems like the license is one-way compatible in the direction needed: https://creativecommons.org/2015/10/08/cc-by-sa-4-0-now-one-way-compatible-with-gplv3/

**Comment by TASagent** [comment]
I really like the idea of a BRB Trivia screen.  And I think using any framework you're comfortable is totally fine.  The existing BrowserSources only use jQuery because I'm inexperienced in webdev and just went for a popular, well-documented tool .  The only notable constraints are it needs to be compatible with OBS's BrowserSource implementation (which I assume is just something like modern chromium), and that it can't be too cpu intensive, since it's going to be rendered by OBS on the streaming computer.  But I think both constraints are _very_ mild.  I have run afoul of the latter before when I tried embedding local mp4's with a `<video>` element.  Perhaps the browser wasn't releasing the resources of prior videos, but after a couple videos OBS would crash with a message saying it was "overloaded".

**Comment by TASagent** [comment]
I'd encourage preparing a Trivia Provider class that's separate from the Trivia Game Engine that will allow someone to swap out trivia databases, or supply (or supplement with) stream-specific or customized trivia.  But the Dependency Injection system should make this relatively straightforward.
